src/csharp/Intel/Iced/Intel/Assembler/AssemblerRegister.cs

```diff
@@ -0,0 +1,271 @@
+/*
+Copyright (C) 2018-2019 de4dot@gmail.com
+
+Permission is hereby granted, free of charge, to any person obtaining
+a copy of this software and associated documentation files (the
+"Software"), to deal in the Software without restriction, including
+without limitation the rights to use, copy, modify, merge, publish,
+distribute, sublicense, and/or sell copies of the Software, and to
+permit persons to whom the Software is furnished to do so, subject to
+the following conditions:
+
+The above copyright notice and this permission notice shall be
+included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+using System;
+using System.Diagnostics;
+
+namespace Iced.Intel
+{
+	/// <summary>
+	/// An assembler register used with <see cref="Assembler"/>.
+	/// </summary>
+	[DebuggerDisplay("{" + nameof(Value) + "}")]
+	public readonly struct AssemblerRegister : IEquatable<AssemblerRegister> {
+
+		/// <summary>
+		/// Creates a new instance.
+		/// </summary>
+		/// <param name="value">A Register</param>
+		public AssemblerRegister(Register value) {
+			Value = value;
+			Flags = AssemblerOperandFlags.None;
+		} 
+
+		/// <summary>
+		/// Creates a new instance.
+		/// </summary>
+		/// <param name="value">A register</param>
+		/// <param name="flags">The mask</param>
+		public AssemblerRegister(Register value, AssemblerOperandFlags flags) {
+			Value = value;
+			Flags = flags;
+		}
+
+		/// <summary>
+		/// The register value.
+		/// </summary>
+		public readonly Register Value;
+
+		/// <summary>
+		/// Gets the mask associated with this register.
+		/// </summary>
+		public readonly AssemblerOperandFlags Flags;
+		
+		/// <summary>
+		/// Apply mask Register K0.
+		/// </summary>
+		public AssemblerRegister k0 => new AssemblerRegister(Value, (Flags & ~AssemblerOperandFlags.RegisterMask) | AssemblerOperandFlags.K0);
```

Comment:
Same here, `k0` can't be used as an opmask register. It just means there's no opmask register.